presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -991,7 +993,26 @@ protected Object visitArrayConstructor(ArrayConstructor node, Object context)
         @Override
         protected Object visitRow(Row node, Object context)
         {
-            throw new PrestoException(NOT_SUPPORTED, "Row expressions not yet supported");
+            RowType rowType = (RowType) expressionTypes.get(node);
+            List<Type> parameterTypes = rowType.getTypeParameters();
+            List<Expression> arguments = node.getItems();
+
+            int cardinality = arguments.size();
+            List<Object> values = new ArrayList<>(cardinality);
+            for (Expression argument : arguments) {
+                Object value = process(argument, context);
+                values.add(value);
+            }
+            if (optimize && hasUnresolvedValue(values)) {
```

Comment:
Why is this `optimize &&`? I think this check should just be `hasUnresolvedValue(values)`